anyway_etl/waze/import_to_db.py

```diff
@@ -0,0 +1,107 @@
+import os
+import dataflows as DF
+import pandas as pd
+import shutil
+from datetime import datetime
+from typing import Callable, Optional
+from sqlalchemy.orm.session import Session
+from pprint import pprint
+from collections import defaultdict
+
+
+from anyway_etl.waze.config import FIELDS, TYPES_MAPPING, QUERIES, \
+    WAZE_TYPE
+from anyway_etl.db import session_decorator
+from anyway_etl.config import ANYWAY_ETL_DATA_ROOT_PATH
+
+
+@session_decorator
+def __does_exist_in_db(session: Session, select_query: str, uuid: str) -> bool:
+    query_with_uuid = select_query.format(uuid=uuid)
+
+    db_execution = session.execute(query_with_uuid)
+
+    results = list(db_execution)
+
+    if len(results) == 0:
+        return False
+
+    count = results[0][0]
+
+    return count > 0
+
+
+@session_decorator
+def __insert_to_db(session: Session, data_type: WAZE_TYPE, row: dict) -> None:
+    session.add(data_type(**row))
+
+
+@session_decorator
+def __update_row(session: Session, data_type: WAZE_TYPE, row: dict) -> None:
+    row['update_time'] = datetime.now()
+
+    uuid = row.get('uuid')
+
+    session.query(data_type). \
+        filter_by(uuid=str(uuid)). \
+        update(row)
+
+
+@session_decorator
```

Comment:
did you test this code? Because it should not work, as each call to session_decorator creates a new session..

so you should reuse the same session in all functions and not use the session_decorator on each function

Reply:
Will fix that!